errbot/plugin_manager.py

```diff
@@ -384,16 +385,31 @@ def update_plugin_places(self, path_list, extra_plugin_dir, autoinstall_deps=Tru
                        for pluginfo in loaded_plugins if pluginfo.error is not None})
         return errors
 
+    def get_all_bot_plugins_ordered(self):
+        all_plugins = []
+        for name in self.plugins_callback_order:
+            # None is a placeholder for any plugin not having a defined order
+            if name is None:
+                all_plugins += [
+                    p for p in self.getPluginsOfCategory(BOTPLUGIN_TAG)
+                    if p.name not in self.plugins_callback_order
+                ]
+            else:
+                p = self.get_plugin_by_name(name)
+                if p is not None:
+                    all_plugins.append(p)
+        return all_plugins
+
     def get_all_active_plugin_objects(self):
         return [plug.plugin_object
-                for plug in self.getPluginsOfCategory(BOTPLUGIN_TAG)
+                for plug in self.get_all_bot_plugins_ordered()
```

Comment:
This also conflate the callback order with the order for everything else and users will start to rely on that.
Please only change the callbackorder at the callback level in https://github.com/errbotio/errbot/blob/master/errbot/core.py#L98